src/main/java/net/mcreator/blockly/java/blocks/TextJoinBlock.java

```diff
@@ -41,8 +41,21 @@
 				return;
 			}
 
+			boolean hasString = false;
+			for (Element element : elements) {
+				if (element != null && BlocklyToCode.directProcessOutputBlock(master, element).matches("\"[^\"]*\"")) {

```

Comment:
directProcessOutputBlock is called twice, plus processOutputBlock, these values should be re-used as this processing can take a while